### PR TITLE
support docker network ls --filter type=XXX

### PR DESCRIPTION
**Pull request by vieux**
Fix https://github.com/docker/swarm/issues/1738

### CODE REVIEW

**Comment by vieux** [comment]
ping @dongluochen @abronan

**Comment by abronan** [comment]
LGTM

@vieux I guess we're ok with `engine-stable` test failing for now until we update after the release right?

**Comment by vieux** [comment]
@abronan I'm going to update the PR to skip the test for engine < 1.10

File: cluster/network.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ type Network struct {
 	Engine *Engine
 }
 
+func (network *Network) isPreDefined() bool {
+	return (network.Name == "none" || network.Name == "host" || network.Name == "bridge")
```
Comment (dongluochen) [inline cluster/network.go]: Shouldn't it be `network.Driver`? 
```
dchen@vm4:~$ ./docker -H vm2:2372 network ls
NETWORK ID          NAME                  DRIVER
d655fe013cf5        devnet                overlay             
12510c2d7c85        help                  overlay             
40a16c3f8697        vm3/docker_gwbridge   bridge              
9722189bb82b        vm3/none              null                
abf0616c3dd0        vm3/bridge            bridge              
```

**Comment by vieux** [comment]
@dongluochen  should be good to merge

File: cluster/network.go (hunk 1 above)
Comment (vieux) [inline cluster/network.go]: @dongluochen no, you can create a network with the bridge driver, as you created it, it's not predefined

**Comment by dongluochen** [comment]
LGTM.

**Pull request merged**